text/0075-span-thread-id.md

```diff
@@ -0,0 +1,35 @@
+- Start Date: 2023-02-15
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/75
+- RFC Status: draft
+
+# Summary
+
+Enrich spans with thread id information.
+
+# Motivation
+
+As we have started to enrich profiling and performance UIs with data from the two products, we realized that we cannot precisely show what spans were executed on each thread (ex in profiling UI) or show what function frames were executed during a span (ex in performance UI). Having span thread id data would allow us to improve the user experience between the two products.
+
+# Supporting Data
+
+In profiling, we currently show the entire span tree regardless of what thread the user is viewing and in performance we default to showing the main thread. This is not very accurate and can lead to misleading our users.
+
+Metrics:
+No usage metrics, but this was requested by customers.
+
+# Options Considered
+
+- Add thread_id to spans
+- Add thread_id and optional ended_thread_id (in case they differ) property to spans
```

Comment:
In OpenTelemetry they refer to this as `thread.id` and `thread.name`: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-thread-attributes

could we use the same naming conventions?

We'll also have to update https://develop.sentry.dev/sdk/performance/span-data-conventions/ with whatever we use.

Reply:
I assume this data will go under `span.data` and not a new top level attribute, which I think is better suited here.

Reply:
Yes, we also need to align here - right now profiling already collects a name as well, so this maybe becomes more a question for performance cc @k-fish, but would you benefit from having thread names available as well?

Reply:
We might be able to use it to annotate the transaction summary to indicate threading happened, so possibly. We don't have any current plans to do that at the moment though.